utils/vim/syntax/swift.vim

```diff
@@ -120,10 +120,8 @@ syn region swiftType contained contains=swiftType,swiftParamDelim
 syn match swiftParamDelim contained
       \ /,/
 " <Generic Clause> (generics)
-syn region swiftTypeParameters contained contains=swiftArchetype,swiftConstraint
+syn region swiftTypeParameters contained contains=swiftVarName,swiftConstraint
       \ start="<" end=">"
-syn match swiftArchetype contained skipwhite nextgroup=swiftTypeDeclaration
```

Comment:
Yeah, though the Archetype was matching the same way as a VarName.  In fact, this is related to the reason for the previously stated "bug".  We don't mark the type decl as being contained (because we have no way to contain it in the function parameter list). A VarName is a pattern of <Name>[: Type].

That said, with the trailing constraints, I think that the lifetime for this matching is limited anyways.